buildbot/check.py

```diff
@@ -23,9 +23,47 @@ def do_check(args):
     ret = True
     return ret
 
+def do_check_sycl_unittests(args):
+    ret = False
+
+    env_tmp=os.environ
+    env_tmp["GTEST_FILTER"] = args.gtest_filter
+
+    pitests = os.path.join("tools", "sycl", "unittests", "pi", "PiTests")
+    unittests_cmd = [pitests]
+    print(unittests_cmd)
+
+    subprocess.check_call(unittests_cmd, cwd=args.obj_dir, env=env_tmp)
+
+    ret = True
+    return ret
+
+def do_check(args):
+    ret = False
+    exceptions = []
+
+    cpu_count = os.cpu_count()
+    if cpu_count is None:
+        cpu_count = DEFAULT_CPU_COUNT
+
+    try:
+        do_check_all_lit(args, cpu_count)
+    except subprocess.CalledProcessError as e:
+        exceptions.append(e)
```

Comment:
Do not fail immediately if a check failed but also enable following checks to run to get a better picture of what fails and what passes.

Reply:
Add this as a design comment

Reply:
Thanks to @AlexeySachkov I figured out how LIT integrates with the unittests which makes this change unnecessary and I will remove it.